File: src/Application Specific/Classes/Entity Collections/InvoiceToBeRaised/InvoiceToBeRaised.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace InvoiceToBeRaiseds

        Public Class InvoiceToBeRaised

            Private _dataTypeValidator As DataTypeValidator
            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

            Private _exists As Boolean = False
            Public Property Exists() As Boolean
                Get
                    Return _exists
                End Get
                Set(ByVal Value As Boolean)
                    _exists = Value
                End Set
            End Property

            Private _deleted As Boolean = False
            Public ReadOnly Property Deleted() As Boolean
                Get
                    Return _deleted
                End Get
            End Property
            Public WriteOnly Property SetDeleted() As Boolean
                Set(ByVal Value As Boolean)
                    _deleted = Value
                End Set
            End Property

#End Region

#Region "InvoiceToBeRaised Properties"


            Private _InvoiceToBeRaisedID As Integer = 0
            Public ReadOnly Property InvoiceToBeRaisedID() As Integer
                Get
                    Return _InvoiceToBeRaisedID
                End Get
            End Property
            Public WriteOnly Property SetInvoiceToBeRaisedID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_InvoiceToBeRaisedID", Value)
                End Set
            End Property


            Private _RaiseDate As DateTime = Nothing
            Public Property RaiseDate() As DateTime
                Get
                    Return _RaiseDate
                End Get
                Set(ByVal Value As DateTime)
                    _RaiseDate = Value
                End Set
            End Property


            Private _InvoiceTypeID As Integer = 0
            Public ReadOnly Property InvoiceTypeID() As Integer
                Get
                    Return _InvoiceTypeID
                End Get
            End Property
            Public WriteOnly Property SetInvoiceTypeID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_InvoiceTypeID", Value)
                End Set
            End Property


            Private _LinkID As Integer = 0
            Public ReadOnly Property LinkID() As Integer
                Get
                    Return _LinkID
                End Get
            End Property
            Public WriteOnly Property SetLinkID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_LinkID", Value)
                End Set
            End Property


            Private _AddressTypeID As Integer = 0
            Public ReadOnly Property AddressTypeID() As Integer
                Get
                    Return _AddressTypeID
                End Get
            End Property
            Public WriteOnly Property SetAddressTypeID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_AddressTypeID", Value)
                End Set
            End Property


            Private _AddressLinkID As Integer = 0
            Public ReadOnly Property AddressLinkID() As Integer
                Get
                    Return _AddressLinkID
                End Get
            End Property
            Public WriteOnly Property SetAddressLinkID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_AddressLinkID", Value)
                End Set
            End Property

            Private _CustomerID As Integer = 0
            Public ReadOnly Property CustomerID() As Integer
                Get
                    Return _CustomerID
                End Get
            End Property
            Public WriteOnly Property SetCustomerID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_CustomerID", Value)
                End Set
            End Property

            Private _TaxRateID As Integer = 0
            Public ReadOnly Property TaxRateID() As Integer
                Get
                    Return _TaxRateID
                End Get
            End Property
            Public WriteOnly Property SetTaxRateID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_TaxRateID", Value)
                End Set
            End Property


            Private _PaymentTermID As Integer = 0
            Public ReadOnly Property PaymentTermID() As Integer
                Get
                    Return _PaymentTermID
                End Get
            End Property
            Public WriteOnly Property SetPaymentTermID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PaymentTermID", Value)
                End Set
            End Property

            Private _PaidByID As Integer = 0
            Public ReadOnly Property PaidByID() As Integer
                Get
                    Return _PaidByID
                End Get
            End Property
            Public WriteOnly Property SetPaidByID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PaidByID", Value)
                End Set
            End Property


#End Region

        End Class

    End Namespace

End Namespace

File: apollo/apollo.fsm/src/Application Specific/Classes/Entity Collections/InvoiceToBeRaised/InvoiceToBeRaised.cs
using System;
using System.Collections;

namespace FSM.Entity
{
    namespace InvoiceToBeRaiseds
    {
        public class InvoiceToBeRaised
        {
            private DataTypeValidator _dataTypeValidator;

            public InvoiceToBeRaised()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            
            

            private int _InvoiceToBeRaisedID = 0;

            public int InvoiceToBeRaisedID
            {
                get
                {
                    return _InvoiceToBeRaisedID;
                }
            }

            public object SetInvoiceToBeRaisedID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_InvoiceToBeRaisedID", value);
                }
            }

            private DateTime _RaiseDate = default;

            public DateTime RaiseDate
            {
                get
                {
                    return _RaiseDate;
                }

                set
                {
                    _RaiseDate = value;
                }
            }

            private int _InvoiceTypeID = 0;

            public int InvoiceTypeID
            {
                get
                {
                    return _InvoiceTypeID;
                }
            }

            public object SetInvoiceTypeID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_InvoiceTypeID", value);
                }
            }

            private int _LinkID = 0;

            public int LinkID
            {
                get
                {
                    return _LinkID;
                }
            }

            public object SetLinkID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_LinkID", value);
                }
            }

            private int _AddressTypeID = 0;

            public int AddressTypeID
            {
                get
                {
                    return _AddressTypeID;
                }
            }

            public object SetAddressTypeID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_AddressTypeID", value);
                }
            }

            private int _AddressLinkID = 0;

            public int AddressLinkID
            {
                get
                {
                    return _AddressLinkID;
                }
            }

            public object SetAddressLinkID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_AddressLinkID", value);
                }
            }

            private int _CustomerID = 0;

            public int CustomerID
            {
                get
                {
                    return _CustomerID;
                }
            }

            public object SetCustomerID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_CustomerID", value);
                }
            }

            private int _TaxRateID = 0;

            public int TaxRateID
            {
                get
                {
                    return _TaxRateID;
                }
            }

            public object SetTaxRateID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_TaxRateID", value);
                }
            }

            private int _PaymentTermID = 0;

            public int PaymentTermID
            {
                get
                {
                    return _PaymentTermID;
                }
            }

            public object SetPaymentTermID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PaymentTermID", value);
                }
            }

            private int _PaidByID = 0;

            public int PaidByID
            {
                get
                {
                    return _PaidByID;
                }
            }

            public object SetPaidByID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PaidByID", value);
                }
            }


            
        }
    }
}